app/Helpers/ContactImportantDateHelper.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Helpers;
+
+use App\Models\ContactImportantDateType;
+use App\Models\Vault;
+use Illuminate\Support\Facades\Cache;
+
+class ContactImportantDateHelper
+{
+    /**
+     * Get the important date type from the vault.
+     *
+     * @return null|ContactImportantDateType
+     */
+    public static function getImportantDateType(string $type, int $vault_id): ?ContactImportantDateType
```

Comment:
I don't understand what this function does.
You put in cache the type? If so, why in an array?
Also, there is no notion of caching in the name of the Helper. Perhaps it would be better to indicate that so it's clear what you are calling.

Reply:
The idea is the get the type only once per request, instead of making xxx calls inside 1 request.
The array cache is only valid for 1 request, and then it's cleared.
I wouldn't put the notion of caching in the name, it's purely technical.